services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/DefaultFileDownloadTest.java

```diff
@@ -58,17 +59,34 @@ public static void tearDown() throws IOException {
         file.delete();
     }
 
-
     @Test
-    void equals_hashcode() {
-        EqualsVerifier.forClass(DefaultFileDownload.class)
-                      .withNonnullFields("completionFuture", "progressFuture", "requestFuture")
-                      .withIgnoredFields("resumableFileDownload", "lock")
-                      .verify();
+    void pause_shouldReturnCorrectly() {
+        CompletableFuture<CompletedFileDownload> future =
+            new CompletableFuture<>();
+        TransferProgress transferProgress = Mockito.mock(TransferProgress.class);
+        GetObjectResponse sdkResponse = getObjectResponse();
+
+        Mockito.when(transferProgress.snapshot()).thenReturn(DefaultTransferProgressSnapshot.builder()
+                                                                                            .bytesTransferred(1000L)
+                                                                                            .sdkResponse(sdkResponse)
+                                                                                            .build());
+
+        DownloadFileRequest request = getDownloadFileRequest();
+
+        DefaultFileDownload fileDownload = new DefaultFileDownload(future,
+                                                                   transferProgress,
+                                                                   () -> request,
+                                                                   null);
+
+        ResumableFileDownload pause = fileDownload.pause();
+        assertThat(pause.downloadFileRequest()).isEqualTo(request);
+        assertThat(pause.bytesTransferred()).isEqualTo(file.length());
+        assertThat(pause.s3ObjectLastModified()).hasValue(sdkResponse.lastModified());
+        assertThat(pause.totalSizeInBytes()).hasValue(sdkResponse.contentLength());
     }
 
     @Test
-    void pause_shouldReturnCorrectly() {
+    void immediatePauseAfterResumeReturnsOriginalRequest() {
```

Comment:
I cannot see any difference between this test case and the previous one after refactoring

Reply:
You're right! Refactoring artifact. Will delete.